src/hooks/api.hook.ts

```diff
@@ -1,7 +1,59 @@
-export function useApi() {
-  async function getSignMessage(): Promise<string> {
-    return 'message_to_sign';
+import { ApiError } from '../dtos/api-error.dto';
+
+export interface ApiInterface {
+  call: <T>(config: CallConfig) => Promise<T>;
+}
+
+export interface CallConfig {
+  url: string;
+  method: 'GET' | 'PUT' | 'POST' | 'DELETE';
+  authenticationToken?: string;
+  data?: any;
+  noJson?: boolean;
+}
+
+export function useApi(): ApiInterface {
+  // TODO (Krysh): add session context here and change if required
+
+  async function call<T>(config: CallConfig): Promise<T> {
+    return fetchFrom<T>(config).catch((error: ApiError) => {
+      if (error.statusCode === 401) {
+        // TODO (Krysh): delete session
+      }
```

Comment:
I would go a little bit of a different route, as re-login would need user interaction because he needs to sign the login message again.

I would delete session and show a "connected but not logged" in state. Need to talk to design about it. As we have no control over the metamask connection, means we can't disconnect from metamask to have some sort of "logout" functionality. But the user can disconnect his metamask from our website at any time

Reply:
Ok, I see. Interesting 🤔